poutyne/framework/model.py

```diff
@@ -334,17 +354,30 @@ def fit_generator(self,
 
         return epoch_iterator.epoch_logs
 
-    def _fit_batch(self, x, y, *, callback=Callback(), step=None, return_pred=False):
-        self.optimizer.zero_grad()
+    def _fit_batch(self,
+                   x,
+                   y,
+                   *,
+                   callback=Callback(),
+                   step=None,
+                   return_pred=False,
+                   zero_all_gradients=True,
+                   do_backprop=True,
+                   batches_per_step=1):
+        if zero_all_gradients:
+            self.optimizer.zero_grad()
 
         loss_tensor, metrics, pred_y = self._compute_loss_and_metrics(x,
                                                                       y,
                                                                       return_loss_tensor=True,
                                                                       return_pred=return_pred)
 
+        loss_tensor = loss_tensor / batches_per_step
```

Comment:
I don't think that the loss that is sent in the logs should be divided. It should only be divided for the backward pass.

Reply:
Also, if we want to support uneven batches, the loss should be multiplied by the batch size. Then, the gradient should be divided by the total number of examples for the optimizer step.

Reply:
I think the same.

Reply:
Fixed by ec1d849831f6d02afc8c658098c0f5db20f4a65e.